src/tools/compiletest/src/lib.rs

```diff
@@ -731,8 +739,13 @@ fn make_test(
     poisoned: &mut bool,
 ) -> Vec<test::TestDescAndFn> {
     let test_path = if config.mode == Mode::RunMake {
-        // Parse directives in the Makefile
-        testpaths.file.join("Makefile")
+        if testpaths.file.join("rmake.rs").exists() {
+            // Parse directives in rmake.rs, which takes precedence over Makefile.
```

Comment:
```suggestion
            // Parse directives in rmake.rs
```

You can't have both at the same time anymore, so there is no precedence.